tests/python/pants_test/backend/python/tasks/native/test_ctypes.py

```diff
@@ -90,39 +91,80 @@ def run_before_task_types(cls):
+        'setup.py': dedent("""\
+        from setuptools import setup, find_packages
+        setup(
+          name='platform_specific_ctypes_cpp_dist',
+          version='0.0.0',
+          packages=find_packages(),
+          data_files=[('', ['libcpp-math-lib.so'])],
+        )
+        """),
       },
     }),
 
   ])
 
   def test_ctypes_c_dist(self):
     platform_specific_dist = self.target_dict['platform_specific_ctypes_c_dist']
-    context, synthetic_target, snapshot_version = self._create_distribution_synthetic_target(
-      platform_specific_dist, extra_targets=[self.target_dict['ctypes_c_library']])
-    self.assertEqual(['platform_specific_ctypes_c_dist==0.0.0+{}'.format(snapshot_version)],
-                      [str(x.requirement) for x in synthetic_target.requirements.value])
-    local_wheel_products = context.products.get('local_wheels')
-    local_wheel = self.retrieve_single_product_at_target_base(
-      local_wheel_products, platform_specific_dist)
-    self.assertTrue(check_wheel_platform_matches_host(local_wheel))
+    self._assert_dist_and_wheel_identity(
+      'platform_specific_ctypes_c_dist', '0.0.0', self.ExpectedPlatformType.current,
+      platform_specific_dist, extra_targets=[self.target_dict['ctypes_c_library']],
+    )
 
   def test_ctypes_cpp_dist(self):
     platform_specific_dist = self.target_dict['platform_specific_ctypes_cpp_dist']
-    context, synthetic_target, snapshot_version = self._create_distribution_synthetic_target(
-      platform_specific_dist, extra_targets=[self.target_dict['ctypes_cpp_library']])
-    self.assertEqual(['platform_specific_ctypes_cpp_dist==0.0.0+{}'.format(snapshot_version)],
-                      [str(x.requirement) for x in synthetic_target.requirements.value])
-
-    local_wheel_products = context.products.get('local_wheels')
-    local_wheel = self.retrieve_single_product_at_target_base(
-      local_wheel_products, platform_specific_dist)
-    self.assertTrue(check_wheel_platform_matches_host(local_wheel))
+    self._assert_dist_and_wheel_identity(
+      'platform_specific_ctypes_cpp_dist', '0.0.0', self.ExpectedPlatformType.current,
+      platform_specific_dist, extra_targets=[self.target_dict['ctypes_cpp_library']],
+    )
+
+  def test_multiplatform_python_setup_resolve_bypasses_python_setup(self):
+    self.set_options_for_scope('python-setup',
+                               platforms=['current', 'linux-x86_64', 'macosx_10_14_x86_64'])
+    platform_specific_dist = self.target_dict['platform_specific_ctypes_cpp_dist']
+    self._assert_dist_and_wheel_identity(
+      'platform_specific_ctypes_cpp_dist', '0.0.0', self.ExpectedPlatformType.current,
+      platform_specific_dist,
+      extra_targets=[self.target_dict['ctypes_cpp_library']],
+    )
+
+  def test_resolve_for_native_sources_allows_current_platform_only(self):
+    platform_specific_dist = self.target_dict['platform_specific_ctypes_cpp_dist']
+    compatible_python_binary_target = self.make_target(
```

Comment:
For the future (not important now), `make_target` is a kind of legacy API that doesn't work well with v2-ifying things; the suggested replacement is to call `self.add_to_build_file` and then `self.target(spec)`